lib/Value/WeBWorK.pm

```diff
@@ -73,6 +73,11 @@ sub Value::Formula::PGseedRandom {
 }
 sub Value::Formula::PGgetRandom {shift->{PGrandom}->random(@_)}
 
+sub Value::maketext {
+	my $envir =eval('$main::rh_envir');
```

Comment:
I think the `eval` in the code you quote is used for error trapping (in case `%main::envir` doesn't exist -- I do testing in perl outside of WeBWorK proper).  I don't think the `eval()` has to do with the Safe compartment issues, here.  But it has been a long time since I worked on this part of the code.